src/client/components/dashboard/network.ts

```diff
@@ -0,0 +1,55 @@
+import { action } from 'mobx'
+import { google } from '../../../shared/proto/messages'
+import { message } from '../../../shared/proto/messages'
+import { Vector2 } from '../../math/vector2'
+import { Network } from '../../network/network'
+import { NUsightNetwork } from '../../network/nusight_network'
+import { RobotModel } from '../robot/model'
+import { DashboardRobotModel } from './dashboard_robot/model'
+import Timestamp = google.protobuf.Timestamp$Properties
+import Overview = message.support.nubugger.Overview
+
+export class DashboardNetwork {
+  public constructor(private network: Network) {
+    this.network.on(Overview, this.onOverview)
+  }
+
+  public static of(nusightNetwork: NUsightNetwork): DashboardNetwork {
+    const network = Network.of(nusightNetwork)
+    return new DashboardNetwork(network)
+  }
+
+  public destroy() {
+    this.network.off()
+  }
+
+  @action
+  private onOverview = (robotModel: RobotModel, overview: Overview) => {
+    const robot = DashboardRobotModel.of(robotModel)
+    const toSeconds = (timestamp: Timestamp | null): number => {
```

Comment:
Can you move this to a non-exported standalone function at the bottom of the file 🙂